apps/webapp/components/shared/referral-hero-banner.tsx

```diff
@@ -0,0 +1,102 @@
+import { IconDownRightArrow } from '@/components/ui/icons'
+import { X } from 'lucide-react'
+import React, { useEffect } from 'react'
+
+interface ReferalHomeBannerProps {
+  onClose: () => void
+  onJoinNow: () => void
+  onHowItWorks: () => void
+}
+
+export default function ReferralHomeBanner({
+  onClose,
+  onJoinNow,
+  onHowItWorks,
+}: ReferalHomeBannerProps) {
+  const [isVisible, setIsVisible] = React.useState(true)
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const scrollPosition = window.scrollY
+      const windowHeight = window.innerHeight
+      const documentHeight = document.documentElement.scrollHeight
+
+      //? Hides banner when user has scrolled past 80% of the page
+      if (scrollPosition > (documentHeight - windowHeight) * 0.8) {
+        setIsVisible(false)
+      } else {
+        setIsVisible(true)
+      }
+    }
+
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [])
+
+  if (!isVisible) return null
+
+  return (
+    <div className="fixed top-0 left-0 right-0 z-50 bg-[#0a0a29] bg-opacity-50 min-h-screen flex items-center justify-center p-4">
+      <div className="bg-gradient-to-br from-[#1a1a4ab4] to-[#0d0d35a0] backdrop-blur-[30px] rounded-3xl p-8 max-w-6xl w-full relative">
+        {/* biome-ignore lint/a11y/useButtonType: <explanation> */}
+        <button
+          className="absolute top-4 right-4 text-[#e728a9] hover:text-[#ff51ed] transition-colors"
+          onClick={onClose}
+        >
+          <X className="w-6 h-6" />
+        </button>
+        <div className="space-y-4 text-center">
+          <p className="text-2xl font-semibold">{content.subtitle}</p>
+          <h1 className="text-4xl font-bold lg:text-7xl sm:text-3xl">{content.title}</h1>
+          <p className="text-4xl font-semibold sm:text-2xl lg:text-6xl">
+            {content.paragraph}
+          </p>
+          {/* biome-ignore lint/style/useSelfClosingElements: <explanation> */}
+          <div className="mx-auto w-[120px] h-[2.37px] bg-[#ff51ed]"></div>
+          <p className="max-w-md mx-auto text-[#7a7ca8] text-lg md:text-xl font-semibold leading-[29px]">
+            {content.paragraph2}
+          </p>
+          <div className="bg-[#0F113E] md:w-1/2 px-16 py-[2.625rem] mx-auto shadow-md bg-gradient-to-br to-cornflowerblue-20 rounded-lg justify-between items-center">
+            {/* biome-ignore lint/a11y/useButtonType: <explanation> */}
+            <button
```

Comment:
Keep in mind to use current components. Do we have a pink (secondary) button variant? If not, we can use the same base and do modifications as needed.

Reply:
Yes we did but its called `accent`. gonna changed